grakn-client/src/main/java/ai/grakn/remote/RemoteGraknTx.java

```diff
@@ -75,27 +88,35 @@ static RemoteGraknTx create(RemoteGraknSession session, GraknTxType txType) {
 
     @Override
     public EntityType putEntityType(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ENTITY);
     }
 
     @Override
     public <V> AttributeType<V> putAttributeType(Label label, AttributeType.DataType<V> dataType) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ATTRIBUTE, var("x").datatype(dataType));
     }
 
     @Override
     public Rule putRule(Label label, Pattern when, Pattern then) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, RULE, var("x").when(when).then(then));
     }
 
     @Override
     public RelationshipType putRelationshipType(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, RELATIONSHIP);
     }
 
     @Override
     public Role putRole(Label label) {
-        throw new UnsupportedOperationException(); // TODO
+        return putSchemaConcept(label, ROLE);
+    }
+
+    private <X extends Concept> X putSchemaConcept(Label label, Schema.MetaSchema meta, VarPattern ... optionalVars){
+        List<VarPattern> vars = new ArrayList<>();
+        vars.add(var("x").label(label).sub(meta.getLabel().getValue()));
```

Comment:
The "more correct" but less nice-looking thing to do is this:
```
vars.add(var("x").label(label).sub(var().label(meta.getLabel())));
```
Then we don't extract the string from the label.

Ideally `sub` should also accept a `Label` though...